### DISCUSSION TITLE
Support for AC200L

### DISCUSSION THREAD

**Discussion by Kathou99** [Patrick762/hassio-bluetti-bt#62] [General]
Hi I try to add the AC200L but it's did not work, I get not entity found did we can make it work? I know it is not in the supported devices

**Comment by Kathou99** [comment]
```
Logger: homeassistant.config_entries
Source: config_entries.py:444
First occurred: 13:56:31 (1 occurrences)
Last logged: 13:56:31

Error setting up entry AC200L2348000450699 for bluetti_bt
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 444, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/bluetti_bt/__init__.py", line 55, in async_setup_entry
    coordinator = PollingCoordinator(hass, address, device_name, polling_interval, persistent_conn, polling_timeout, max_retries)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/bluetti_bt/coordinator.py", line 153, in __init__
    bluetti_device = build_device(address, device_name)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/bluetti_mqtt/bluetooth/__init__.py", line 28, in build_device
    if match[1] == 'AC200M':
       ~~~~~^^^
TypeError: 'NoneType' object is not subscriptable

```

**Comment by Patrick762** [comment]
Is the bluetooth name starting with PBOX or the device name "AC200L"?

**Reply by Kathou99** [reply]
yes

**Reply by Kathou99** [reply]
![Screenshot_20240223_142127_Gmail](https://github.com/Patrick762/hassio-bluetti-bt/assets/71993497/6df83542-5ec0-4c5b-8bf4-6b1b6ee98ad0)

**Reply by Patrick762** [reply]
Could you run the bluetti-logger described [here](https://github.com/warhammerkid/bluetti_mqtt?tab=readme-ov-file#reverse-engineering)?
This is needed to find out which version of communication protocol was used.

**Reply by Kathou99** [reply]
hi, I try to run it but I got a lot of issue/error showing, did the addon Bluetti to MQTT in HA will work?

I got this when I put the MAC adress of the bluetti 

```
Waiting for connection...
Waiting for connection...
Waiting for connection...
ERROR:root:Error connecting to device A0:A3:B3:0D:17:1A:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/bleak/backends/bluezdbus/client.py", line 204, in connect
    reply = await self._bus.call(
  File "/usr/local/lib/python3.10/site-packages/dbus_fast/aio/message_bus.py", line 371, in call
    await future
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/bluetti_mqtt/bluetooth/client.py", line 78, in _connect
    await self.client.connect()
  File "/usr/local/lib/python3.10/site-packages/bleak/__init__.py", line 531, in connect
    return await self._backend.connect(**kwargs)
  File "/usr/local/lib/python3.10/site-packages/bleak/backends/bluezdbus/client.py", line 141, in connect
    async with async_timeout(timeout):
  File "/usr/local/lib/python3.10/site-packages/async_timeout/__init__.py", line 129, in __aexit__
    self._do_exit(exc_type)
  File "/usr/local/lib/python3.10/site-packages/async_timeout/__init__.py", line 212, in _do_exit
    raise asyncio.TimeoutError
asyncio.exceptions.TimeoutError
Waiting for connection...
Waiting for connection...
Waiting for connection...
Waiting for connection...
Waiting for connection...
Waiting for connection...
Waiting for connection...
Waiting for connection...
Waiting for connection...
Waiting for connection...
Waiting for connection...
ERROR:root:Error connecting to device A0:A3:B3:0D:17:1A:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/bleak/backends/bluezdbus/client.py", line 204, in connect
    reply = await self._bus.call(
  File "/usr/local/lib/python3.10/site-packages/dbus_fast/aio/message_bus.py", line 371, in call
    await future
asyncio.exceptions.CancelledError
```

**Reply by Kathou99** [reply]
The result I am so far
```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/00-banner.sh

-----------------------------------------------------------
 Add-on: Bluetti to MQTT
 MQTT bridge between Bluetti and Home Assistant
-----------------------------------------------------------
 Add-on version: 1.4.1
 You are running the latest version of this add-on.
 System: Home Assistant OS 11.5  (amd64 / qemux86-64)
 Home Assistant Core: 2024.2.3
 Home Assistant Supervisor: 2024.02.0
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
cont-init: info: /etc/cont-init.d/00-banner.sh exited 0
cont-init: info: running /etc/cont-init.d/01-log-level.sh
cont-init: info: /etc/cont-init.d/01-log-level.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
[14:10:13] INFO: Reading configuration settings...
[14:10:14] INFO: Starting bluetti-logger...
[14:10:14] INFO: Messages are NOT published to the MQTT broker in logger mode.
Traceback (most recent call last):
  File "/usr/local/bin/bluetti-logger", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/bluetti_mqtt/logger_cli.py", line 119, in main
    asyncio.run(log(args.address, args.log))
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.10/site-packages/bluetti_mqtt/logger_cli.py", line 55, in log
    devices = await check_addresses({address})
  File "/usr/local/lib/python3.10/site-packages/bluetti_mqtt/bluetooth/__init__.py", line 55, in check_addresses
    return [build_device(d.address, d.name) for d in filtered]
  File "/usr/local/lib/python3.10/site-packages/bluetti_mqtt/bluetooth/__init__.py", line 55, in <listcomp>
    return [build_device(d.address, d.name) for d in filtered]
  File "/usr/local/lib/python3.10/site-packages/bluetti_mqtt/bluetooth/__init__.py", line 28, in build_device
    if match[1] == 'AC200M':
TypeError: 'NoneType' object is not subscriptable
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

**Reply by Patrick762** [reply]
Do you have your Bluetooth dongle near your Powerstation? That could explain the error.

If I find some time next week I'll try to add the discovery to the integration (maybe also with a better output format).

**Reply by Kathou99** [reply]
yes is like 1-2ft from the power station

**Reply by wi-cheese** [reply]
Hi all-

I'm going to chime in here. Sometime late last year I was able to successfully get Bluetti-BT communicating with HA and my Bluetti AC200Max. It's been working very well and I appreciate Patrick's hard work on developing the tools. I use Bluetti-BT every day in order to capture my solar production statistics!

Just today I got a new AC200L, so I was hoping to add it to HA as well. Since it's been several months since I set up my AC200Max with Bluetti-BT/HA, I kind of forgot how I did it. If I recall, once I got the ESP32 set up as a Bluetooth proxy and I had the Bluetti-BT integration installed, I think HA automatically discovered my AC200Max.

So where I'm at now is that I have my new AC200L turned on and I ensured that my phone isn't connected to the AC200L via BT/Bluetti app since I know only one device can connect via BT at a time. When I go into the Bluetti-BT integration and click Add Device, it shows the AC200L (with SN after it) amongst the other BT devices in my house, so that's a good sign. However, when I select the AC200L device and choose Submit, it states "No entities or devices" and "Failed to set up" on the homescreen of the Bluetti-BT integration.

So I'm not sure if this is because the AC200L isn't supported for Bluetti-BT or if I'm just doing something wrong. I'm happy to provide more detail if needed in order to help troubleshooting getting the AC200L to work with Bluetti-BT, but I'm fairly new to HA and might need some guidance :-)

Jeff

**Reply by Kathou99** [reply]
So far, I know he is not supported yet. This issue is made to try to support this unit AC200L :)

**Reply by wi-cheese** [reply]
Okay, thanks. I just sponsored Patrick762 because Bluetti-BT has been great for the AC200Max and I hope he can get it working for the AC200L soon.

**Reply by Patrick762** [reply]
@wi-cheese Thank you for sponsoring the proect.
Could you run the discovery script mentioned above?

**Reply by Patrick762** [reply]
In case it's really the same protocol as the AC200M, I can just copy the config over. Are there major differences in input or output ports?

**Reply by Kathou99** [reply]
I don't think it's has a lot of difference but we could start with this and if something need to be change we will tell you? 🫠

**Reply by wi-cheese** [reply]
You're welcome.

As far as input/output port differences between the AC200Max and the AC200L, there really aren't many differences. The AC200L is mostly just an upgrade of the AC200Max, including a new screen, upgraded components, andthe addition of wi-fi management capability (via the Bluetti app). I believe there's one different DC output and one different input.

This new AC200L model also has three different UPS modes, which would be very beneficial to be able to change/control through HA. That function is available via Bluetooth and wi-fi from the Bluetti app.

I'm happy to help gather needed information, but I'm a newbie when it comes to running commands and diagnostics in HA. So the discovery script you're referring to, is that the "$ bluetti-discovery --scan" portion of the Warrhammerkid link above? If so, I'll have to figure out how to run commands in SSH. I have OpenSSH installed on HA, I started it in HA, then went into the web GUI for it. I got to the "[core-ssh ~]$" prompt, but it doesn't recognize the "$ bluetti-discovery --scan" command. I'm guessing I'm missing a step somewhere.

One other thing to note is that there's a pending firmware update for all of the components on my new AC200L. Should I update the firmware or wait until we get this discovery figured out? I didn't want to update the firmware yet since I saw a post about Bluetti-BT communication with a firmware update on a different model.

Thanks!
Jeff

**Reply by wi-cheese** [reply]
@Patrick762 , what do you mean that you can copy the config over? Is there something I can help with to test the AC200L?

Thanks!

**Comment by wi-cheese** [comment]
Patrick, do you think you can look at this sometime soon? I'd still really
like to get the AC200L working with this and I've been trying to wait
patiently ;-)

Thanks,
Jeff

**Comment by user150aa346** [comment]
Patrick,

I have the AC200L and I'm able to get basic data with the add-on you created. Thanks.

Mike

**Comment by albaga3** [comment]
I have a Bluetti AC200L.
I run Home Assistant with Docker Compose, and the integration connects correctly to the Bluetti.
The Home Assistant version is 2025.5.1 and the HACS version is 1.34.0.

I installed Home Assistant OS version 2025.5.2 and HACS version 2.0.5, but in this case, the integration doesn't work.

Any idea what might be wrong?

Thanks.
Best regards.

**Reply by bisonbet** [reply]
I'm seeing this problem as well - I cannot see my AC200L, but I do see a different system
